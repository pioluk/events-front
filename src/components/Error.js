import React from 'react'

const styles = {
  root: {
    fontSize: 12,
    color: 'rgb(244, 67, 54)'
  }
}

const Error = ({ message }) =>
  <div style={styles.root}>
    {message}
  </div>

export default Error
