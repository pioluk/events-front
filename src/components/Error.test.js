// global describe,it,expect

import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Error from './Error'

describe('Error component', () => {
  it('should have correct styles and structure', () => {
    const wrapper = shallow(<Error />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should contain passed message', () => {
    const message = 'This field is required.'
    const wrapper = shallow(<Error message={message} />)
    expect(wrapper.children().length).toEqual(1)
    expect(wrapper.contains(message)).toBe(true)
  })
})
