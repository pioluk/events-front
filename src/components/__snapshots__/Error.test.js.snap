exports[`Error component should have correct styles and structure 1`] = `
<div
  style={
    Object {
      "color": "rgb(244, 67, 54)",
      "fontSize": 12,
    }
  } />
`;
